.github/workflows/run-tests.yml

```diff
@@ -17,24 +17,38 @@ jobs:
     steps:
       - name: Checkout repository
         uses: actions/checkout@v3
-      - name: Checkout actions
+      - name: Run tests
+        run: |
+          cargo test --lib transport
+      - name: Cancel workflow runs for commit on error
+        if: failure()
+        uses: ./.github/.release/actions/actions/utils/fast-jobs-failure
+  acceptance-tests:
+    name: Acceptance tests
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout project
+        uses: actions/checkout@v3
+      - name: Checkout mock-server action
         uses: actions/checkout@v3
         with:
           repository: pubnub/client-engineering-deployment-tools
           ref: v1
           token: ${{ secrets.GH_TOKEN }}
           path: .github/.release/actions
-      # TODO: setup features for contract tests
-      - name: Run tests
+      - name: Run mock server action
+        uses: ./.github/.release/actions/actions/mock-server
+        with:
+          token: ${{ secrets.GH_TOKEN }}
+          features-path: tests/features
+      - name: Run acceptance tests
         run: |
-          cargo test 
-      - name: Cancel workflow runs for commit on error
-        if: failure()
-        uses: ./.github/.release/actions/actions/utils/fast-jobs-failure
+          cargo test --test contract
```

Comment:
Shouldn't we have test modules e.g. `publish`, `subscribe`, etc. instead of just `contract`?

Reply:
Exploring more on this.
As per My limited understanding how cargo test works - We need one entry point file with same name. And for cucumber settings one file starts execution of all tests. We can filter it out via tags later on. But the entry point/ file.rs is one.